docs/excel/custom-functions-parameter-options.md

```diff
@@ -28,31 +27,83 @@ Whereas regular parameters are required, optional parameters are not. When a use
  * @returns {number} The sum of the numbers.
  */
 function add(first, second, third) {
-  if (third !== undefined) {
-    return first + second + third;
+   if (third === null) {
+    third=0;
   }
-  return first + second;
+  return first + second + third;
 }
 CustomFunctions.associate("ADD", add);
 ```
 
-When you define a function that contains one or more optional parameters, you should specify what happens when the optional parameters are undefined. In the following example, `zipCode` and `dayOfWeek` are both optional parameters for the `getWeatherReport` function. If the `zipCode` parameter is undefined, the default value is set to `98052`. If the `dayOfWeek` parameter is undefined, it is set to Wednesday.
+#### [TypeScript](#tab/typescript)
+
+```typescript
+/**
+ * Add two numbers
+ * @customfunction 
+ * @param first First number.
+ * @param second Second number.
+ * @param [third] Third number to add. If omitted, third = 0.
+ * @returns The sum of the numbers.
+ */
+function add(first:number, second:number, third?:number):number {
+  if (third === null) {
+    third=0;
+  }
+  return first + second + third;
+}
+CustomFunctions.associate("ADD", add);
+```
+
+---
+
+> [!NOTE]
+> Excel will pass optional parameters with a null value. This means default-initialized parameters in TypeScript will not work as expected. Therefore don't use `function add(first:number, second:number, third=0):number` syntax because it will not initialize third to 0. Instead use the TypeScript syntax as shown in the previous example.
```

Comment:
```suggestion
> When no value is specified for an optional parameter, Excel assigns it the value `null`. This means default-initialized parameters in TypeScript will not work as expected. Therefore, don't use the syntax `function add(first:number, second:number, third=0):number` because it will not initialize `third` to 0. Instead use the TypeScript syntax as shown in the previous example.
```